gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java

```diff
@@ -71,9 +72,10 @@ public LocalFsSpecProducer(Config config) {
 
   private Future<?> writeSpec(Spec spec, SpecExecutor.Verb verb) {
     if (spec instanceof JobSpec) {
-      String specString = spec.toString();
+      URI specUri = spec.getUri();
       // format the JobSpec to have file of <flowGroup>_<flowName>.job
-      String jobFileName = specString.replace('/', '_').substring(specString.lastIndexOf(':')+2, specString.length()-1) + ".job";
+      String[] uriTokens = specUri.getPath().split("/");
```

Comment:
We can be more DRY(Don't Repeat Yourself) by making a small function named `getJobFileName(uri)`